File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.java
class Solution {
    public int maxFrequencyElements(int[] nums) {
        int a[]=new int[101];
        for(int i:nums)
        {
            a[i]++;
        }
        int maxi=0;
        for(int i=0;i<=100;i++)
        {
            maxi=Math.max(maxi,a[i]);
        }
        int ans=0;
        for(int i=0;i<=100;i++)
        {
            if(a[i]==maxi)
            {
                ans+=maxi;
            }
        }
        return ans;
    }
    
}

File: 3005-count-elements-with-maximum-frequency/3005-count-elements-with-maximum-frequency.cpp
class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        
        int maxi=0;
        int n=nums.size();
        unordered_map<int,int> m;
        for(int i=0;i<n;++i)
        {
            m[nums[i]]++;
        }
        
        for(int i=0;i<n;++i)
        {
            if(m[nums[i]]>maxi)
                maxi=m[nums[i]];
        }
        int ans=0;
        for(int i=0;i<n;++i)
        {
            if(m[nums[i]]==maxi)
                ans++;
        }
        
        
        return ans;
        
    }
};